src/mpyl/projects/versioning.py

```diff
@@ -0,0 +1,149 @@
+"""Versioning and upgrade utilities for mpyl projects.
+### Writing an upgrade script
+
+To write an upgrade script, create a new class that inherits from `Upgrader` and
+implements the `upgrade` method. This class should then be added to the `UPGRADERS`
+list in this module.
+
+"""
+import copy
+import pkgutil
+from abc import ABC
+from pathlib import Path
+from typing import Optional, Generator
+
+from deepdiff import DeepDiff
+from ruamel.yaml.compat import ordereddict
+
+from ..utilities.yaml import yaml_to_string, load_for_roundtrip
+
+VERSION_FIELD = "mpylVersion"
+
+
+def get_releases() -> list[str]:
+    embedded_releases = pkgutil.get_data(__name__, "releases/releases.txt")
+    if not embedded_releases:
+        raise ValueError("File releases/releases.txt not found in package")
+    releases = embedded_releases.decode("utf-8").strip().splitlines()
+    return list(reversed(releases))
+
+
+def get_latest_release() -> str:
+    return get_releases()[0]
+
+
+class Upgrader(ABC):
+    """Base class for upgrade scripts"""
+
+    target_version: str
+
+    def upgrade(self, previous_dict: ordereddict) -> ordereddict:
+        return previous_dict
+
+
+class Upgrader11(Upgrader):
```

Comment:
What would be the function names if we switch major versions? For example `2.0.1` and at some point we might reach `2.0.11`

Reply:
@SamTheisens
[[TECH-556] Rename to be future proof](https://github.com/Vandebron/mpyl/pull/207/commits/924f1f39544ceb164db80986d12b7609b50499de)